meta_update_middleware/template-queries.go

```diff
@@ -0,0 +1,283 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/couchbase/gocb/v2"
+)
+
+// init runs before main() is evaluated
+func init() {
+	log.Println("template-queries:init()")
+}
+
+func getModels(conn CbConnection, dataset string, app string, doctype string, subDocType string) (jsonOut []string) {
+	log.Println("getModels(" + dataset + "," + app + "," + doctype + "," + subDocType + ")")
+	start := time.Now()
+
+	fileContent, err := os.ReadFile("sqls/getModels.sql")
+	if err != nil {
+		log.Fatal(err)
+	}
+	tmplGetModelsSQL := string(fileContent)
+	tmplGetModelsSQL = strings.Replace(tmplGetModelsSQL, "{{vxDBTARGET}}", conn.vxDBTARGET, -1)
+	tmplGetModelsSQL = strings.Replace(tmplGetModelsSQL, "{{vxDOCTYPE}}", doctype, -1)
+	tmplGetModelsSQL = strings.Replace(tmplGetModelsSQL, "{{vxSUBDOCTYPE}}", subDocType, -1)
+
+	models_requiring_metadata := queryWithSQLStringSA(conn.Scope, tmplGetModelsSQL)
+
+	log.Println(fmt.Sprintf("\tin %v", time.Since(start)))
+	return models_requiring_metadata
+}
```

Comment:
Was there a reason not to use Go's templating engine here? If not, it could be useful to consider in the future. 

https://pkg.go.dev/text/template

Reply:
Will postpone this to later then